### PR TITLE
Custom grammar for markdown support

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This adds our current functionality and fixes some issues. Quote blocks are parse but not formatted yet.

I suggest looking at the commits individually, the biggest diff is from adding the compiled parser.

We might be able to use a webpack loader instead of compiling the parser. I can look into that if we want.

### CODE REVIEW

File: shared/common-adapters/markdown.js [hunk 1]
```diff
@@ -37,165 +34,59 @@ const codeSnippetBlockStyle = {
...
-    const matchingMark = m[0]
-    return {matchingMark, restText: text.slice(matchingMark.length)}
+function createComponent (type, key, children) {
+  switch (type) {
+    case 'inline-code':
+      return <Text type='Body' key={key} style={codeSnippetStyle}>{children}</Text>
+    case 'code-block':
+      return <Text type='Body' key={key} style={codeSnippetBlockStyle}>{children}</Text>
+    case 'text':
+      return <Text type='Body' key={key} style={{color: undefined, fontWeight: undefined}}>{children}</Text>
```
Comment (chrisnojima) [inline shared/common-adapters/markdown.js]: make sure to get my styles form https://github.com/keybase/client/pull/5388. we should make these style props consts so we don't get perf hits from making the objects over and over

File: shared/common-adapters/markdown.js [hunk 2]
```diff
@@ -86,18 +86,22 @@ function process (ast) {
...
-        } else if (typeof child === 'string') {
-          acc.push([child])
         } else {
-          acc.push(child.component)
+          if (typeof child === 'string') {
+            acc.push([child])
+          } else {
+            acc.push(child.component)
+          }
         }
```
Comment (chromakode) [inline shared/common-adapters/markdown.js]: :+1:

File: shared/common-adapters/markdown.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {emojiIndex} from 'emoji-mart'
 import Emoji from './emoji'
 import React, {PureComponent} from 'react'
 import {globalStyles, globalColors, globalMargins} from '../styles'
-import parser from '../markdown/parser.pegjs'
+import parser from '../markdown/parser'
```
Comment (chromakode) [inline shared/common-adapters/markdown.js]: :+1: commit looks good

File: shared/markdown/parser.pegjs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+ = StrikeMarker children:FromStrike* StrikeMarker { return {type: 'strike', children: children}; }
+
+CodeBlock
+ = Ticks3 children:InsideCodeBlock* Ticks3 { return {type: 'code-block', children}; }
+
+InlineCode
+ = Ticks1 children:InsideInlineCode* Ticks1 { return {type: 'inline-code', children}; }
+
+Emoji
+ = EmojiMarker children:(InsideEmojiMarker / "::")* EmojiMarker { return {type: 'emoji', children: [children.join('')]}; }
```
Comment (chromakode) [inline shared/markdown/parser.pegjs]: Does/should this also include unicode emoji? We should parse them so we can use the `Emoji` component on them, right?

**Pull request merged**